micro_sam/training/sam_trainer.py

```diff
@@ -216,12 +222,13 @@ def _compute_iterative_loss(self, batched_inputs, y_one_hot, num_subiter, multim
             iou_regression_loss += net_iou_regression_loss
             mean_model_iou += net_mean_model_iou
 
-            # Determine the next prompts based on current predictions.
-            with torch.no_grad():
-                # Get the mask and logit predictions corresponding to the predicted object
-                # (per actual object) with the best IOU.
-                masks, logits = self._get_best_masks(batched_outputs, batched_iou_predictions)
-                batched_inputs = self._update_prompts(batched_inputs, y_one_hot, masks, logits)
+            if num_subiter > 1:
```

Comment:
We could be more efficient here, and check if we still have a next subiteration to go for which we need the masks and updated prompts. Then normal training would also profit from this.
So something like 
```python
if i < (num_subiter - 1):
...
```
(I think that's correct, but please double check that it's not off by one!)

Reply:
Ahha yes, makes sense. Thanks for the mention!